### PR TITLE
Ports and environment variables can be set through a project's config file

### PR DESCRIPTION
**Pull request by JakeTheSnake3p0**
Lean projects can now be configured to expose additional ports as required by simply updating the project's config.json file. The impetus in my case was the fact that Tensorboard was inaccessible in my local research environment.

I'm unfamiliar with how this repository's code fits together well enough to know if this is the best spot in which to place the update. Currently it only affects the `lean research` command but not others.

### CODE REVIEW

File: README.md
Comment (jmerle) [inline README.md:1002]: This part of the readme is automatically generated by the `scripts/readme.py` script based on the `--help` messages of each command, so the next time that script is ran this piece of documentation will be removed. Project configuration is documented on [https://www.lean.io/docs/lean-cli/user-guides/configuration#04-Project-Configuration](https://www.lean.io/docs/lean-cli/user-guides/configuration#04-Project-Configuration), which is generated from [https://github.com/QuantConnect/Documentation/blob/master/05%20Lean%20CLI/02%20User%20Guides/02%20Configuration/04%20Project%20Configuration.html](https://github.com/QuantConnect/Documentation/blob/master/05%20Lean%20CLI/02%20User%20Guides/02%20Configuration/04%20Project%20Configuration.html). I recommend moving the documentation for `docker-ports` and `docker-env` to that file.

File: lean/commands/research.py
Comment (jmerle) [inline lean/commands/research.py:168]: I recommend moving this logic to [`LeanRunner.get_basic_docker_config()`](https://github.com/QuantConnect/lean-cli/blob/master/lean/components/docker/lean_runner.py#L159), that way these new project configuration properties can be used with `lean backtest`, `lean live`, `lean optimize`, and `lean research` (instead of only `lean research`).

File: lean/commands/research.py
Comment (jmerle) [inline lean/commands/research.py:161]: nit - should use double quotes for consistency with the rest of the project

File: lean/commands/research.py
Comment (jmerle) [inline lean/commands/research.py:168]: nit - should use double quotes for consistency with the rest of the project

**Comment by JakeTheSnake3p0** [comment]
@jmerle I've made the updates suggested.

**Review by jmerle — APPROVED** [review]
LGTM :+1:, assuming the documentation will be updated after this PR is merged.

**Comment by JakeTheSnake3p0** [comment]
Documentation has been [updated](https://github.com/QuantConnect/Documentation/pull/541) and awaiting merge.

**Comment by JakeTheSnake3p0** [comment]
@jmerle I think my recent branch merge reverted a workflow progress. Status is sitting at 'awaiting approval'.

**Comment by jaredbroad** [comment]
Sorry for the delay @JakeTheSnake3p0 - solid PR; merging! 🎉

**Pull request merged**
